### PR TITLE
Clarify ResponseInit options

### PR DESCRIPTION
**Pull request by kflansburg**
In https://github.com/cloudflare/workers-rs/issues/567 it was noted that our docs do not show all of the possible options for `ResponseInit`. This updates our docs to reflect what is in `worker-types`.

### CODE REVIEW

File: content/workers/runtime-apis/response.md [hunk 1]
```diff
@@ -60,8 +63,6 @@ Valid options for the `options` object include: {{<definitions>}}
   - A getter to get the body contents.
 - `response.bodyUsed` {{<type>}}boolean{{</type>}}
   - A boolean indicating if the body was used in the response.
-- `response.encodeBody` {{<type>}}string{{</type>}}
```
Comment (kflansburg) [inline content/workers/runtime-apis/response.md:63]: `workers-types` indicates that this is not actually a property of `Request`

File: content/workers/runtime-apis/response.md [hunk 2]
```diff
@@ -39,16 +39,19 @@ let response = new Response(body, init);
 
 Valid options for the `options` object include: {{<definitions>}}
 
+<!-->Does `cf` do anything in a Response?<-->
```
Comment (kflansburg) [inline content/workers/runtime-apis/response.md]: Do we have more information on the possible values of `cf`?

Comment (himikof) [reply content/workers/runtime-apis/response.md]: Unfortunately, there indeed seems to be no documentation on what options are supported in `ResponseInit.cf` and what do they actually do.
Also, I'm not sure about `any | null` typing: the [C++ implementation in workerd](https://github.com/cloudflare/workerd/blob/bb8c8cd75e6a85dc88a1905553eb1ce4e1e58576/src/workerd/api/http.h#L1007-L1008) clearly expects `object | null`. Might be an issue of insufficient typing [here](https://github.com/cloudflare/workerd/blob/bb8c8cd75e6a85dc88a1905553eb1ce4e1e58576/src/workerd/api/http.h#L1015-L1018), leading to a problem in `workers-types`.

File: content/workers/runtime-apis/response.md [hunk 3]
```diff
@@ -39,16 +39,19 @@ let response = new Response(body, init);
 
 Valid options for the `options` object include: {{<definitions>}}
 
+<!-- Does `cf` do anything in a Response? -->
```
Comment (irvinebroque) [inline content/workers/runtime-apis/response.md]: Not sure quite what you mean?

Comment (kflansburg) [reply content/workers/runtime-apis/response.md]: I don't know what configuring `cf` in the `Response` object does. What features are supported? We have this well documented for `Request`. Am I missing something?

Comment (irvinebroque) [reply content/workers/runtime-apis/response.md]: I'm not aware of any case where setting the value of the `cf` property on a response changes behavior

File: content/workers/runtime-apis/response.md (hunk 2 above)
Comment (jasnell) [reply content/workers/runtime-apis/response.md]: I would see if we can handle documenting `cf` more adequately in a separate PR.

File: content/workers/runtime-apis/response.md (hunk 3 above)
Comment (jasnell) [reply content/workers/runtime-apis/response.md]: That is correct. Setting the `cf` property on the `Response` is largely informational. Someone could take it, make modifications to it, and pass that on to a subsequent subrequest but setting the value when constructing the `Response` has no functional impact other than to set the value of the `response.cf` property.

**Review by jasnell — APPROVED** [review]

**Pull request merged**